src/components/common/entityData/RelatedEntityCard/cmp.tsx

```diff
@@ -5,10 +5,14 @@ import { StyledRelatedEntityCard } from './styles'
 
 export const RelatedEntityCard = ({
   children,
+  disabled = false,
   onClick: handleClick,
 }: RelatedEntityCardProps) => {
   return (
-    <StyledRelatedEntityCard onClick={handleClick}>
+    <StyledRelatedEntityCard
+      onClick={() => !disabled && handleClick()}
```

Comment:
You are passing a new function reference on each render, causing unnecessary re-renders. Check the disabled status inside the handleClick function (and use useCallback for it).
This is a tiny performance issue that doesn't matter too much, but many of them together can impact performance.